apis/compute/v1alpha1/serverset_types.go

```diff
@@ -134,19 +134,28 @@ type ServerSetObservation struct {
 	ReplicaStatuses []ServerSetReplicaStatus `json:"replicaStatus,omitempty"`
 }
 
-// ServerSetReplicaStatus are the observable fields of a ServerSetReplicaStatus.
+// ServerSetReplicaStatus contains the status of a Server Replica.
 type ServerSetReplicaStatus struct {
 	xpv1.ResourceStatus `json:",inline"`
 	// +kubebuilder:validation:Enum=ACTIVE;PASSIVE
-	Role Role   `json:"role"`
-	Name string `json:"name"`
+	Role        Role                 `json:"role"`
+	Name        string               `json:"name"`
+	Index       int                  `json:"index"`
+	NICStatuses []ServerSetNICStatus `json:"nicStatus,omitempty"`
 	// +kubebuilder:validation:Enum=UNKNOWN;READY;ERROR
 	Status string `json:"status"`
 	// ErrorMessage relayed from the backend.
 	ErrorMessage string      `json:"errorMessage,omitempty"`
 	LastModified metav1.Time `json:"lastModified,omitempty"`
 }
 
+// ServerSetNICStatus contains the status of a Server's Nic.
+type ServerSetNICStatus struct {
```

Comment:
can't se reuse the nic status from nic_types?

Reply:
We could, however we might expose more information than we need and we will somehow tight couple the ServerSet to the Nic

Reply:
We decided to go with the existing NicStatus